### DISCUSSION TITLE
Bugs with TextCenterVertical() ?

### DISCUSSION THREAD

**Discussion by chandra-arifin** [CommunityToolkit/Maui.Markup#96] [Q&A]
Code 1:
`Label lblDate = new Label()
 {
        VerticalTextAlignment = TextAlignment.Center,
 }.Text("Date");`

and            

Code 2:
`Label lblDate = new Label().TextCenterVertical().Text("Date");`

i have 2 code, code 1 runs correctly, but code 2 the text not vertically center. is it a bug? or i write the wrong extension?

i try TextCenter(), but nothing works.

**Comment by bijington** [comment]
@chandra-arifin are you able to provide more detail here. I have created the following example which works as expected:

```csharp
using CommunityToolkit.Maui.Markup;

namespace Markup;

public class MainPage : ContentPage
{
  public MainPage()
  {
    Content = new StackLayout
    {
      Children =
      {
        new Label
	{
	  Text = "Without Extension",
	  HeightRequest = 200,
	  BackgroundColor = Colors.PaleGoldenrod,
	  VerticalTextAlignment = TextAlignment.Center
	},

	new Label()
        {
          HeightRequest = 200,
          BackgroundColor = Colors.DarkOrange,
        }
	  .TextCenterVertical()
	  .Text("With")
      }
    };
  }
}
```

This gives the result of

<img width="901" alt="Screenshot 2022-08-01 at 14 01 53" src="https://user-images.githubusercontent.com/17139988/182154030-bb24e543-e42e-47d6-8e15-8a9df9692484.png">

To give a further explanation of `TextCenterVertical` vs `TextCenterHorizontal` vs `TextCenter`. Basically `TextCenter` applies both `TextCenterVertical` and `TextCenterHorizontal`.

I suspect you are wanting the Label to be positioned centrally rather than just its text? If so then you want to take a look at the [`CenterVertical`](https://docs.microsoft.com/en-gb/dotnet/communitytoolkit/maui/markup/extensions/view-extensions#centervertical) method.

**Comment by chandra-arifin** [comment]
I use grid. 2 column.
Column 1 is label. And column 2 is Entry.

Label is not vertically center (not center as Entry component). Entry component is just entry, without text vertically center.

Tommorow i will paste my code here.


Get Outlook for Android<https://aka.ms/ghei36>
________________________________

**Comment by CliffAgius** [comment]
Closed and answered
